File: 99.育成中コンテンツ/広告/アドフリくん/adfurikunSDK_Android_moviereward_3.20.0/SampleJava/app/src/main/java/sample/adfurikunsdk/jp/samplejava/SimpleBanner.java
package sample.adfurikunsdk.jp.samplejava;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.TextView;

import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunMovieError;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunBanner;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunBannerAdInfo;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunBannerLoadListener;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunBannerVideoListener;

/**
 * バナー広告用サンプル
 */
public class SimpleBanner extends Activity {

    private TextView mTitle;
    private TextView mDescription;
    private FrameLayout mBannerViewFrame;
    private Button mLoad;
    private Button mPlay;

    /**
     * バナー広告インスタンス
     */
    private AdfurikunBanner mBanner;
    /**
     * バナー広告情報
     */
    private AdfurikunBannerAdInfo mBannerInfo;

    /**
     * 広告読み込みリスナー
     */
    private AdfurikunBannerLoadListener mLoadListener = new AdfurikunBannerLoadListener() {
        @Override
        public void onBannerLoadFinish(AdfurikunBannerAdInfo adInfo, String appId) {
            // 広告の準備完了直後に実行する処理を記述してください
            Utils.log("onBannerLoadFinish appId = " + appId);
            if (adInfo != null) {
                mBannerInfo = adInfo;
                String title = adInfo.getTitle();
                String description = adInfo.getDescription();
                mTitle.setText(title);
                mDescription.setText(description);
                mPlay.setEnabled(true);
            }
        }

        @Override
        public void onBannerLoadError(AdfurikunMovieError error, String appId) {
            // 広告の準備失敗直後に実行する処理を記述してください
            String errorType = "";
            if (error != null) {
                errorType = error.getErrorType().name();
            }
            Utils.log("onBannerLoadError appId = " + appId + ", errorType = " + errorType);
        }
    };

    /**
     * 広告再生リスナー
     */
    private AdfurikunBannerVideoListener mVideoListener = new AdfurikunBannerVideoListener() {
        @Override
        public void onBannerViewPlayStart(String appId) {
            // 広告の再生開始直後に実行する処理を記述してください
            Utils.log("onBannerViewPlayStart appId = " + appId);
        }

        @Override
        public void onBannerViewPlayFinish(String appId, boolean isVideoAd) {
            // 広告の再生が完了したら実行する処理を記述してください
            Utils.log("onBannerViewPlayFinish appId = " + appId + ", isVideoAd = " + isVideoAd);
        }

        @Override
        public void onBannerViewPlayFail(String appId, AdfurikunMovieError error) {
            // 広告の再生に失敗した時に実行する処理を記述してください
            String errorType = "";
            if (error != null) {
                errorType = error.getErrorType().name();
            }
            Utils.log("onBannerViewPlayFail appId = " + appId + ", errorType = " + errorType);
        }

        @Override
        public void onBannerViewClicked(String appId) {
            // 広告の画面をクリックした時に実行する処理を記述してください
            Utils.log("onBannerViewClicked appId = " + appId);
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_banner);
        setTitle(getIntent().getStringExtra("title"));

        mDescription = findViewById(R.id.description);
        mTitle = findViewById(R.id.title);
        mBannerViewFrame = findViewById(R.id.banner_view_frame);
        mLoad = findViewById(R.id.load);
        mPlay = findViewById(R.id.play);

        // ボタンを押すと広告を読み込む
        mLoad.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mBanner != null) {
                    mBanner.load();
                }
            }
        });

        // ボタンを押すと広告を表示する
        mPlay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mBanner != null) {
                    mBanner.play();
                }
            }
        });

        // バナー広告のインスタンスを生成する
        // 広告枠IDとActivityを引数で渡してください
        mBanner = new AdfurikunBanner(this, AdfurikunConst.ADFURIKUN_BANNER_APPID,
                Utils.convertDpToPx(this, 320),
                Utils.convertDpToPx(this, 50));
        mBanner.setAdfurikunBannerLoadListener(mLoadListener);
        mBanner.setAdfurikunBannerVideoListener(mVideoListener);

        mBannerViewFrame.addView(mBanner.getBannerView());
    }

    @Override
    protected void onDestroy() {
        if (mBanner != null) {
            mBanner.onDestroy();
        }
        super.onDestroy();
    }
}

File: 99.育成中コンテンツ/広告/アドフリくん/adfurikunSDK_Android_moviereward_3.20.0/SampleKotlin/app/src/main/java/jp/glossom/sample/samplekotlin/SimpleBanner.kt
package jp.glossom.sample.samplekotlin

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.FrameLayout
import android.widget.TextView
import jp.tjkapp.adfurikunsdk.moviereward.*

/**
 * バナー広告用サンプル
 */
class SimpleBanner : Activity() {

    private lateinit var mTitle: TextView
    private lateinit var mDescription: TextView
    private lateinit var mBannerViewFrame: FrameLayout
    private lateinit var mLoad: Button
    private lateinit var mPlay: Button

    /** バナー広告インスタンス */
    private var mBanner: AdfurikunBanner? = null
    /** バナー広告情報 */
    private var mBannerInfo: AdfurikunBannerAdInfo? = null

    /** 広告読み込みリスナー */
    private val mLoadListener = object : AdfurikunBannerLoadListener {
        override fun onBannerLoadFinish(adInfo: AdfurikunBannerAdInfo?, appId: String?) {
            // 広告の準備完了直後に実行する処理を記述してください
            Utils.log("onBannerLoadFinish appId = $appId")
            adInfo?.let {
                mBannerInfo = it
                mTitle.text = it.title
                mDescription.text = it.description
                mPlay.isEnabled = true
            }
        }

        override fun onBannerLoadError(error: AdfurikunMovieError?, appId: String?) {
            // 広告の準備失敗直後に実行する処理を記述してください
            Utils.log("onBannerLoadError appId = $appId, errorType = ${error?.errorType?.name}")
        }
    }

    /** 広告再生リスナー */
    private val mVideoListener = object : AdfurikunBannerVideoListener {
        override fun onBannerViewPlayStart(appId: String?) {
            // 広告の再生開始直後に実行する処理を記述してください
            Utils.log("onBannerViewPlayStart appId = $appId")
            mPlay.isEnabled = false
        }

        override fun onBannerViewPlayFinish(appId: String?, isVideoAd: Boolean) {
            // 広告の再生に失敗した時に実行する処理を記述してください
            Utils.log("onBannerViewPlayFinish appId = $appId, isVideoAd = $isVideoAd")
        }

        override fun onBannerViewPlayFail(appId: String?, error: AdfurikunMovieError?) {
            // 広告の再生が完了したら実行する処理を記述してください
            Utils.log("onBannerViewPlayFail appId = $appId, errorType = ${error?.errorType?.name}")
        }

        override fun onBannerViewClicked(appId: String?) {
            // 広告の画面をクリックした時に実行する処理を記述してください
            Utils.log("onBannerViewClicked appId = $appId")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_banner)
        title = intent.getStringExtra("title")

        mTitle = findViewById(R.id.title)
        mDescription = findViewById(R.id.description)
        mBannerViewFrame = findViewById(R.id.banner_view_frame)
        mLoad = findViewById(R.id.load)
        mPlay = findViewById(R.id.play)

        // ボタンを押すと広告を読み込む
        mLoad.setOnClickListener {
            mBanner?.load()
        }

        // ボタンを押すと広告を表示する
        mPlay.setOnClickListener {
            mBanner?.play()
        }

        // バナー広告のインスタンスを生成する
        // 広告枠IDとActivityを引数で渡してください
        mBanner = AdfurikunBanner(
            this, AdfurikunConst.ADFURIKUN_BANNER_APPID,
            Utils.convertDpToPx(this, 320),
            Utils.convertDpToPx(this, 50)
        )
        mBanner?.setAdfurikunBannerLoadListener(mLoadListener)
        mBanner?.setAdfurikunBannerVideoListener(mVideoListener)

        mBannerViewFrame.addView(mBanner?.bannerView)
    }

    override fun onDestroy() {
        mBanner?.onDestroy()
        super.onDestroy()
    }
}
